python/ray/data/_internal/execution/streaming_executor.py

```diff
@@ -291,8 +296,27 @@ def _scheduling_loop_step(self, topology: Topology) -> bool:
         update_operator_states(topology)
 
         # Update the progress bar to reflect scheduling decisions.
-        for op_state in topology.values():
+        log_str = "Execution Progress Status:\n"
+        cur_time = time.perf_counter()
+        for op, op_state in topology.items():
             op_state.refresh_progress_bar()
+            if tqdm is not None:
+                log_str += (
+                    tqdm.tqdm.format_meter(
+                        op_state.num_completed_tasks,
+                        op.num_outputs_total(),
+                        cur_time - self._start_time,
+                        prefix=op.name,
+                        bar_format="{desc:<35.34}{percentage:3.0f}%|{bar}{r_bar}",
+                    )
+                    + "\n"
+                )
+            else:
+                log_str += (
+                    f"{op.name}: "
+                    f"{op_state.num_completed_tasks}/{op.num_outputs_total()}\n"
+                )
```

Comment:
I think it's okay to not format with tqdm.  the graphics are not super useful. It is `state.summary_str()` that matters the most. (I'm not sure if `summary_str` already contains the `{op_state.num_completed_tasks}/{op.num_outputs_total()}` progress. If not, also log it).

I just found that we already have a similar function `_debug_dump_topology(topology)`. 
So I think we can considate them. By default, dump the info into data log here. If the flag is enabled, dump the info into both stdout and data log in the `while op is not None:` loop.